Resources/public/js/views/services/ez-locationviewviewservice.js

```diff
@@ -43,6 +44,150 @@ YUI.add('ez-locationviewviewservice', function (Y) {
         },
 
         /**
+         * _sendContentToTrashConfirmBox event handler on send to trash button
+         *
+         * @method _sendContentToTrashConfirmBox
+         * @protected
+         * @param {Object} e event facade of the sendToTrashAction event
+         */
+        _sendContentToTrashConfirmBox: function (e) {
+            e.preventDefault();
+            this.fire('confirmBoxOpen', {
+                config: {
+                    title: "Are you sure you want to send this content to trash?",
+                    confirmHandler: Y.bind(function () {
+                        this._sendToTrash(e);
+                    }, this),
+                    cancelHandler: Y.bind(function () {
+                    }, this),
+                },
+            });
+        },
+
+        /**
+         * Sends location to trash, triggering loading parent location and notifications
+         *
+         * @method _sendToTrash
+         * @protected
+         * @param {Object} e event facade of the sendToTrashAction event
+         */
+        _sendToTrash: function (e) {
+            var tasks, parentLocation, endLoadParent,
+                that = this,
+                location = this.get('location'),
+                locationId = location.get('id'),
+                contentName = e.content.get('name');
+
+            this._sendContentToTrashNotificationStarted(locationId, contentName);
+
+            tasks = new Y.Parallel();
```

Comment:
indeed you don't need to use Y.Parallel here. Y.Parallel is useful when you want to run several asynchronous operations at the same time and do something when they are all done, here you can indeed call `location.trash` in the callback of `_loadParent`.